dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/OpenAIImageGeneration.cs

```diff
@@ -34,50 +45,55 @@ public class OpenAIImageGeneration : OpenAIClientBase, IImageGeneration
     ) : base(httpClient, logger)
     {
         Verify.NotNullOrWhiteSpace(apiKey);
-        this.HTTPClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);
+        this._authorizationHeaderValue = $"Bearer {apiKey}";
+        this._organizationHeaderValue = organization;
+    }
 
-        if (!string.IsNullOrEmpty(organization))
+    /// <summary>Adds headers to use for OpenAI HTTP requests.</summary>
+    private protected override void AddRequestHeaders(HttpRequestMessage request)
+    {
+        base.AddRequestHeaders(request);
+
+        request.Headers.Add("Authorization", this._authorizationHeaderValue);
+        if (!string.IsNullOrEmpty(this._organizationHeaderValue))
         {
-            this.HTTPClient.DefaultRequestHeaders.Add("OpenAI-Organization", organization);
+            request.Headers.Add("OpenAI-Organization", this._organizationHeaderValue);
```

Comment:
wouldn't it be faster doing it once in the ctor, using DefaultRequestHeaders, as opposed to checking if the value is not empty on each request?

Reply:
As is noted in the PR description, we can't use DefaultRequestHeaders.  If we're accepting an HttpClient from a caller, adding to DefaultRequestHeaders will end up setting that header for any other requests they make on that same HttpClient, which makes other use of that instance a bug.  It's also not thread-safe to modify DefaultRequestHeaders while it might be in use, so if someone is making calls on that HttpClient while this instance is being constructed and mutating DefaultRequestHeaders, that's a bug.  From a perf perspective, it's also a wash, as all of the headers need to be copied from DefaultRequestHeaders to these request headers, with various checks and enumerations and whatnot along the way.

Reply:
I agree, I think this has been answered. Resolving!